apps/sms/services/js/messaging/messaging_client.js

```diff
@@ -51,18 +50,69 @@
       });
     },
 
+    /* Sends an SMS based on the specified parameter object.
+     *
+     * TODO: Splitting this complicated method
+     * @param {object} opts Options object that contains :-
+     * - recipients (string or array of string): contains the list of
+     *   recipients for this message
+     * - content (string): the message's body
+     * - serviceId (optional long or string): the SIM serviceId we use to send
+     *   the message
+     * - onsuccess (optional function): will be called when one SMS has been
+     *   sent successfully, with the request's result as argument. Can be called
+     *   several times.
+     * - onerror (optional function): will be called when one SMS transmission
+     *   failed, with the error object as argument. Can be called several times.
+     * - oncomplete (optional function): will be called when all messages have
+     *   been sent. It's argument will have the following properties:
+     *   + hasError (boolean): whether we had at least one error
+     *   + return (array): each item is an object with the following properties:
+     *     . success (boolean): whether this is a success or an error
+     *     . result (request's result): the request's result object
+     *     . recipient (string): the recipient used for this transmission.
+     */
```

Comment:
Also could you please add ```returns``` directive to the methods in this file? All they return ```{Promise<..something>}```. Thanks!